lib/concurrent/dataflow.rb

```diff
@@ -50,12 +50,15 @@ def dataflow_with!(executor, *inputs, &block)
   end
   module_function :dataflow_with!
 
-  private 
+  private
 
   def call_dataflow(method, executor, *inputs, &block)
     raise ArgumentError.new('an executor must be provided') if executor.nil?
     raise ArgumentError.new('no block given') unless block_given?
-    raise ArgumentError.new('not all dependencies are IVars') unless inputs.all? { |input| input.is_a? IVar }
+    not_ivars = inputs.map { |input| !input.is_a? IVar }
+    unless not_ivars.empty?
```

Comment:
Yes, that should have been a `select`, it was right locally but not committed. I'm updating to not allocate an array on every run, will preserve the original logic.